src/webapp/components/action-list-table/ActionListTable.tsx

```diff
@@ -366,18 +369,18 @@ export const ActionListTable: React.FC<ActionListTableProps> = props => {
                 text: i18n.t("Import JSON translations"),
                 icon: <Icon>translate</Icon>,
                 onClick: () => {
-                    translationImportRef.current?.startImport();
+                    startImport();
                 },
             },
         ],
-        [openImportDialog, translationImportRef]
+        [openImportDialog, startImport]
     );
 
     return (
         <PageWrapper>
             {dialogProps && <ConfirmationDialog isOpen={true} maxWidth={"xl"} {...dialogProps} />}
 
-            <ImportTranslationDialog type="action" ref={translationImportRef} onSave={handleTranslationUpload} />
+            {ImportTranslationDialog}
```

Comment:
The consensus in the community seems to be that hooks should not return components or views; they should concern themselves with logic. So your previous approach was correct.